src/CC.hs

```diff
@@ -0,0 +1,138 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+-- |
+--
+-- Utility for formatting @'Idea'@ data in accordance with the Code Climate
+-- spec: <https://github.com/codeclimate/spec>
+--
+module CC
+    ( printIssue
+    , fromIdea
+    ) where
+
+import Data.Aeson (ToJSON(..), (.=), encode, object)
+import Data.Char (toUpper)
+import Data.Monoid ((<>))
+import Data.Text (Text)
+import Language.Haskell.Exts.SrcLoc (SrcSpan(..))
+
+import qualified Data.Text as T
+import qualified Data.ByteString.Lazy.Char8 as C8
+
+import Idea (Idea(..), Severity(..))
+
+data Issue = Issue
+    { issueType :: Text
+    , issueCheckName :: Text
+    , issueDescription :: Text
+    , issueContent :: Text -- TODO: Markdown type?
+    , issueCategories :: [Text]
+    , issueLocation :: Location
+    , issueRemediationPoints :: Int
+    }
+
+data Location = Location FilePath Position Position
+data Position = Position Int Int
+
+instance ToJSON Issue where
+    toJSON Issue{..} = object
+        [ "type" .= issueType
+        , "check_name" .= issueCheckName
+        , "description" .= issueDescription
+        , "content" .= object
+            [ "body" .= issueContent
+            ]
+        , "categories" .= issueCategories
+        , "location" .= issueLocation
+        , "remediation_points" .= issueRemediationPoints
+        ]
+
+instance ToJSON Location where
+    toJSON (Location path begin end) = object
+        [ "path" .= path
+        , "positions" .= object
+            [ "begin" .= begin
+            , "end" .= end
+            ]
+        ]
+
+instance ToJSON Position where
+    toJSON (Position line column) = object
+        [ "line" .= line
+        , "column" .= column
+        ]
+
+-- | Print an @'Issue'@ with trailing null-terminator and newline
+--
+-- The trailing newline will be ignored, but makes the output more readable
+--
+printIssue :: Issue -> IO ()
+printIssue = C8.putStrLn . (<> "\0") . encode
+
+-- | Convert an hlint @'Idea'@ to a datatype more easily serialized for CC
+fromIdea :: Idea -> Issue
+fromIdea Idea{..} = Issue
+    { issueType = "issue"
+    , issueCheckName = "HLint/" <> T.pack (camelize ideaHint)
+    , issueDescription = T.pack ideaHint
+    , issueContent = content ideaFrom ideaTo
```

Comment:
You are dropping ideaNote which should definitely end up in the content - those are warnings about when refactorings have subtle side conditions.

Reply:
I didn't realize that; could you offer a rough template of what content should look like, taking that into account?